cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ConfigurationFinder.java

```diff
@@ -130,18 +131,24 @@ private static URI findTemplates(Path home) {
       return templatesFolderPath.toUri();
     }
 
-    // 2 create/use new template sets folder
+    // 2. use template jar
+    File jarFile = TemplatesJarUtil.getJarFile(false, templatesFolderPath.getParent().toFile());
+    if (jarFile != null) {
+      return jarFile.toPath().toUri();
+    }
```

Comment:
Please use Path instead of File.
For the if condition you can use Files.exists(jarFile)

Reply:
This would cause a NullPointerException, since getJarFile returns null if nothing was found.
We could change the return value of getJarFile, but then we also need to adjust some other classes that use the method.

Reply:
Not if you do it like jarFile != null && Files.exists(jarFile).
Btw. shall we keep the deprecated annotation of getjarFile?

Reply:
Not sure if I get your intention. `Files.exists` works only on `Path` not on `File`. So again we had to change the return type of `getJarFile`, which I wanted to avoid.

> Btw. shall we keep the deprecated annotation of getjarFile?
I am not sure if this would be correct right now, as the old structure is still supported, so we may need to continue using `getJarFile` somewhere else